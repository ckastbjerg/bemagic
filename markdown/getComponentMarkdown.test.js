/* eslint-env jest */
const dummyBlock = require('../dummyBlock');
const getComponentMarkdown = require('./getComponentMarkdown');

test('it works', () => {
    global.config = { namespace: 'prefix' };
    expect(getComponentMarkdown({
        component: Object.assign({}, dummyBlock),
        exampleMarkup: null,
    })).toMatchSnapshot();
});
