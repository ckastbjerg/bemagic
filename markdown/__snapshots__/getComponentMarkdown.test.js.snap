exports[`test it works 1`] = `
"
        
        ## Block \`block\`
        
    
        
        ###### Table of contents
        - **[Theming](#theming)**
        - **[Modifiers](#modifiers)**
        - **[States](#states)**
        - **[Elements](#elements)**
        - **[Component overview](#component-overview)**
        - **[Contributing](#contributing)**
    
        
        
        <a name=Usage></a>
        #### Usage
    
        <div class=\"block\">…</div>
    
        
        
        <a name=Theming></a>
        #### Theming
    
        The \`test\` supports [\`theming\`](../CONVENTIONS.md) and can change its
        appearance when paired with the \`prefix-undefined\` class.
        Note however, that themes may apply only for parts of the component.

        \`\`\`html
        <div class=\"test prefix-undefined\">…</div>
        <!-- or using nesting -->
        <div class=\"prefix-undefined\">
            <div class=\"test\">…</div>
        </div>
        \`\`\`
        <sub>See the [Component Overview](#component-overview) for more details.</sub>
    
        
        
        <a name=modifier></a>
        #### Modifier
    ;
        
            The block component makes use of
            [\`modifiers\`](../CONVENTIONS.md#modifiers) to alter its appearance
        .

        \`\`\`html
        <div class=\"block block--modifier\">…</div>
        \`\`\`

        #####All component \`modifier\`

        
        | Name | Description |
        |:-----|:-----|

        
                | **\`modifier\`**
                | Some description... |
            
    
        <sub>See the [Component Overview](#component-overview) for more details</sub>
    
        
        
        <a name=states></a>
        #### States
    ;
        
            The block component makes use of
            [\`modifiers\`](../CONVENTIONS.md#modifiers) to alter its appearance
        .

        \`\`\`html
        undefined
        \`\`\`

        #####All component \`states\`

        
        | Name | Description |
        |:-----|:-----|

        
                | **\`dimmed\`**
                | Some description... |
            
    
        <sub>See the [Component Overview](#component-overview) for more details</sub>
    
        
        
        <a name=elements></a>
        #### Elements
    ;
        
            The block component makes use of
            [\`modifiers\`](../CONVENTIONS.md#modifiers) to alter its appearance
        .

        \`\`\`html
        <div class=\"block__element\">…</div>
        \`\`\`

        #####All component \`elements\`

        
        | Name | Description |
        |:-----|:-----|

        
                | **\`element\`**
                | Some description... |
            
    
        <sub>See the [Component Overview](#component-overview) for more details</sub>
    
        
        <a name=Component Overview></a>
        #### Component Overview
    | Class Name | Type | States | Tag |
|:-----|:-----|:-----|:-----|

        | **[\`block\`](#block)**
        | [component](../CONVENTIONS.md#components)
        | None
        | N/A
    
        | **[\`block block--modifier\`](#blockblock--modifier)**
        | [modifier](../CONVENTIONS.md#modifiers)
        | None
        | N/A
    
        | **[\`block__element\`](#block__element)**
        | [element](../CONVENTIONS.md#elements)
        | None
        | N/A
    
        | **[\`block__element block__element--modifier\`](#block__elementblock__element--modifier)**
        | [modifier](../CONVENTIONS.md#modifiers)
        | None
        | N/A
    


        
        <a name=block></a>
        #### \`block\` (required)
        > Some description...
        <div class=\"block\">…</div>
        Elements using this class will change appearance when used inside [theme]
        (../CONVENTIONS.md#theming) \`inverse\`.
        The component will react to \`hover\`. states
        ###### Specifications

        - **Tag:** div,,- **Required:** true
        ----------
    
        
        <a name=blockblock--modifier></a>
        #### \`block block--modifier\` (required)
        > Some description...
        <div class=\"block block--modifier\">…</div>
        Elements using this class will change appearance when used inside [theme]
        (../CONVENTIONS.md#theming) \`inverse\`.
        The component will react to \`hover\`. states
        ###### Specifications

        - **Tag:** div,,- **Required:** true
        ----------
    
        
                
        <a name=block__element></a>
        #### \`block__element\` 
        > Some description...
        <div class=\"block__element\">…</div>
        Elements using this class will change appearance when used inside [theme]
        (../CONVENTIONS.md#theming) \`inverse\`.
        The component will react to \`hover\`. states
        ###### Specifications

        - **Tag:** div,,- **Required:** false
        ----------
    
                
        <a name=blockblock--modifier></a>
        #### \`block block--modifier\` (required)
        > Some description...
        <div class=\"block block--modifier\">…</div>
        Elements using this class will change appearance when used inside [theme]
        (../CONVENTIONS.md#theming) \`inverse\`.
        The component will react to \`hover\`. states
        ###### Specifications

        - **Tag:** div,,- **Required:** true
        ----------
    
            
        
        
        <a name=Contributing></a>
        #### Contributing
    ;
        
            Please view the **[contribution guidelines](../CONVENTIONS.md)** before
            modifying this component
        .
    
    "
`;
